client/pkg/eventlog/eventlog_impl.go

```diff
@@ -460,6 +499,38 @@ func (r *logReader) Read(ctx context.Context, size int16) ([]*ce.Event, error) {
 	return events, nil
 }
 
+func (r *logReader) SyncReadStream(ctx context.Context, size int16) ([]*ce.Event, error) {
+	if r.cur == nil {
+		segment, err := r.elog.selectReadableSegment(ctx, r.pos)
+		if errors.Is(err, errors.ErrOffsetOnEnd) {
+			r.elog.refreshReadableSegments(ctx)
+			segment, err = r.elog.selectReadableSegment(ctx, r.pos)
+		}
+		if err != nil {
+			return nil, err
+		}
+		r.cur = segment
+	}
+
+	events, err := r.cur.SyncReadStream(ctx, r.pos, size, uint32(r.pollingTimeout(ctx)))
+	if err != nil {
+		if errors.Is(err, errors.ErrOffsetOverflow) {
+			r.elog.refreshReadableSegments(ctx)
+			if r.switchSegment(ctx) {
+				return nil, errors.ErrTryAgain
+			}
+		}
+		return nil, err
+	}
+
+	r.pos += int64(len(events))
+	if r.pos == r.cur.EndOffset() {
+		r.switchSegment(ctx)
```

Comment:
does here need to retry?

Reply:
This error of EndOffset does not need to be retried